test/integration/desktop/gnome.nix

```diff
@@ -0,0 +1,112 @@
+let
+  common = import ../common.nix;
+  inherit (common) users paretoPatchedDash dashboard displayManager ssh;
+in {
+  name = "GNOME";
+
+  nodes.dashboard = {
+    imports = [
+      (dashboard {})
+    ];
+  };
+
+  nodes.gnome = {
+    pkgs,
+    lib,
+    ...
+  }: {
+    imports = [
+      (users {})
+      (paretoPatchedDash {inherit pkgs lib;})
+      (displayManager {inherit pkgs;})
+    ];
+
+    virtualisation.resolution = {
+      x = 800;
+      y = 600;
+    };
+
+    services.xserver.enable = true;
+    services.xserver.desktopManager.gnome.enable = true;
+
+    environment.systemPackages = [pkgs.gnomeExtensions.appindicator];
+    services.udev.packages = [pkgs.gnome-settings-daemon];
+
+    # Enable AppIndicator extension automatically
+    programs.dconf.profiles.user.databases = [
+      {
+        settings = {
+          "org/gnome/shell" = {
+            enabled-extensions = ["appindicatorsupport@rgcjonas.gmail.com"];
+          };
+        };
+      }
+    ];
+
+    environment.gnome.excludePackages = with pkgs; [
+      atomix # puzzle game
+      cheese # webcam tool
+      epiphany # web browser
+      evince # document viewer
+      geary # email reader
+      gedit # text editor
+      gnome-characters
+      gnome-music
+      gnome-photos
+      gnome-terminal
+      gnome-tour
+      hitori # sudoku game
+      iagno # go game
+      tali # poker game
+      totem # video player
+    ];
+  };
+
+  interactive.nodes.gnome = {...}:
+    ssh {port = 2221;} {};
+
+  enableOCR = true;
+
+  testScript = ''
+    # Test setup
+    # terminal.succeed("su - alice -c 'mkdir -p /home/alice/.config'")
+    for m in [gnome, dashboard]:
+      m.systemctl("start network-online.target")
+      m.wait_for_unit("network-online.target")
+
+
+    # Test 4: Test the tray icon
+    gnome.wait_for_x()
+    for unit in [
+        'paretosecurity-trayicon',
+        'paretosecurity-user',
+        'paretosecurity-user.timer'
+    ]:
+        status, out = gnome.systemctl("is-enabled " + unit, "alice")
+        assert status == 0, f"Unit {unit} is not enabled (status: {status}): {out}"
+
+    # GNOME system tray is in the top-right corner
+    gnome.succeed("xdotool mousemove 670 10")
+    gnome.screenshot("1-mouse-at-670-10.png")
+    # gnome.succeed("xdotool click 1")
```

Comment:
remove?

Reply:
Not in review ready state yet. 

I had a more polished version, but it failed in CI. Then I reverted to the less polished version that I saved as the "first ugly passing version", but is failing too. 

I noticed it is also not reliable locally. But nix caching builds once it succeeds makes you think "oh this works now", but then you change the test name and get a failure.